core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/ccm/upgrade/CcmUpgradeFlowConfig.java

```diff
@@ -0,0 +1,118 @@
+import com.sequenceiq.flow.core.config.FlowFinalizerCallback;
+import com.sequenceiq.flow.core.config.RetryableFlowConfiguration;
+
+@Component
+public class CcmUpgradeFlowConfig extends AbstractFlowConfiguration<CcmUpgradeState, CcmUpgradeEvent>
+        implements RetryableFlowConfiguration<CcmUpgradeEvent> {
+
+    private static final List<Transition<CcmUpgradeState, CcmUpgradeEvent>> TRANSITIONS =
+            new Builder<CcmUpgradeState, CcmUpgradeEvent>()
+                    .defaultFailureEvent(CCM_UPGRADE_FAILURE_EVENT)
+                    .from(INIT_STATE).to(CCM_UPGRADE_PREPARATION_STATE)
+                    .event(CCM_UPGRADE_EVENT)
+                    .defaultFailureEvent()
+
+                    .from(CCM_UPGRADE_PREPARATION_STATE).to(CCM_UPGRADE_RE_REGISTER_TO_CP)
+                    .event(CCM_UPGRADE_PREPARATION_FINISHED_EVENT)
+                    .failureEvent(CCM_UPGRADE_PREPARATION_FAILED_EVENT)
+
+                    .from(CCM_UPGRADE_RE_REGISTER_TO_CP).to(CCM_UPGRADE_REMOVE_AUTOSSH)
+                    .event(CCM_UPGRADE_RE_REGISTER_FINISHED_EVENT)
+                    .defaultFailureEvent()
+
+                    .from(CCM_UPGRADE_REMOVE_AUTOSSH).to(CCM_UPGRADE_UNREGISTER_HOSTS)
+                    .event(CCM_UPGRADE_REMOVE_AUTOSSH_FINISHED_EVENT)
+                    .defaultFailureEvent()
+
+                    .from(CCM_UPGRADE_UNREGISTER_HOSTS).to(CCM_UPGRADE_FINISHED_STATE)
+                    .event(CCM_UPGRADE_UNREGISTER_HOSTS_FINISHED_EVENT)
+                    .defaultFailureEvent()
+
+                    .from(CCM_UPGRADE_FINISHED_STATE).to(FINAL_STATE)
+                    .event(FINALIZED_EVENT)
+                    .defaultFailureEvent()
+
+                    .build();
+
+    private static final FlowEdgeConfig<CcmUpgradeState, CcmUpgradeEvent> EDGE_CONFIG = new FlowEdgeConfig<>(INIT_STATE, FINAL_STATE,
+            CCM_UPGRADE_FAILED_STATE, FAIL_HANDLED_EVENT);
+
+    @Inject
+    private StackStatusFinalizer stackStatusFinalizer;
+
+    public CcmUpgradeFlowConfig() {
+        super(CcmUpgradeState.class, CcmUpgradeEvent.class);
+    }
+
+    @Override
+    public FlowTriggerCondition getFlowTriggerCondition() {
+        return getApplicationContext().getBean(CcmUpgradeFlowTriggerCondition.class);
+    }
+
+    @Override
+    protected List<Transition<CcmUpgradeState, CcmUpgradeEvent>> getTransitions() {
+        return TRANSITIONS;
+    }
+
+    @Override
+    protected FlowEdgeConfig<CcmUpgradeState, CcmUpgradeEvent> getEdgeConfig() {
+        return EDGE_CONFIG;
+    }
+
+    @Override
+    public CcmUpgradeEvent[] getEvents() {
+        return CcmUpgradeEvent.values();
+    }
+
+    @Override
+    public CcmUpgradeEvent[] getInitEvents() {
+        return new CcmUpgradeEvent[] {
+                CCM_UPGRADE_EVENT
+        };
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Cluster's connectivity manager upgrade";
```

Comment:
Nit: `Cluster's connectivity manager` -> `Cluster Connectivity Manager` or simply `CCM`. (I am not sure the flow name is visible to the user. If so, the official expanded name should be used instead of the acronym.)

Reply:
ok